src/program/snabb_lwaftr/setup.lua

```diff
@@ -47,10 +49,16 @@ function lwaftr_app(c, conf)
       prepend(postprocessing_apps_v6, "egress_filterv6")
    end
 
-   set_preprocessors(c, preprocessing_apps_v4, "lwaftr.v4")
-   set_preprocessors(c, preprocessing_apps_v6, "lwaftr.v6")
-   set_postprocessors(c, "lwaftr.v6", postprocessing_apps_v6)
-   set_postprocessors(c, "lwaftr.v4", postprocessing_apps_v4)
+   set_preprocessors(c, preprocessing_apps_v4, "icmpechov4.south")
+   config.link(c, "icmpechov4.north -> lwaftr.v4")
+   config.link(c, "lwaftr.v4 -> icmpechov4.north")
+
+   set_preprocessors(c, preprocessing_apps_v6, "icmpechov6.south")
+   config.link(c, "icmpechov6.north -> lwaftr.v6")
+   config.link(c, "lwaftr.v6 -> icmpechov6.north")
+
+   set_postprocessors(c, "icmpechov6.south", postprocessing_apps_v6)
+   set_postprocessors(c, "icmpechov4.south", postprocessing_apps_v4)
```

Comment:
I would prefer if you use the preprocessors and postprocessors logic, otherwise is going to be confuse. Setting the preprocessors and postprocessors to the lwaftr should be the last step. I think you want something like this:

```
append(preprocessing_apps_v4, { name = "icmpechov4", input = "south", output = "north" })
prepend(postprocessing_apps_v4, { name = "icmpechov4", input = "south", output = "north" })
append(preprocessing_apps_v6, { name = "icmpechov6", input = "south", output = "north" })
prepend(postprocessing_apps_v6, { name = "icmpechov6", input = "south", output = "north" })

set_preprocessors(c, preprocessing_apps_v4, "lwaftr.v4")
set_preprocessors(c, preprocessing_apps_v6, "lwaftr.v6")
set_postprocessors(c, "lwaftr.v6", postprocessing_apps_v6)
set_postprocessors(c, "lwaftr.v4", postprocessing_apps_v4)
```

And probably there's a bug in the `set_preprocessors` and `set_postprocessors` methods as they assume the name of the `input` and `output` links is always _input_ and _output_. This change should fix it:

```
function set_preprocessors(c, apps, dst)
   assert(type(apps) == "table")
   link_apps(c, apps)
   local last_app, output = apps[#apps], "output"
   if type(last_app) == "table" then
      last_app, output = last_app["name"], last_app["output"]
   end
   config.link(c, ("%s.%s -> %s"):format(last_app, output, dst))
end
```

Similarly for `set_postprocessors`.